server/client.go

```diff
@@ -879,8 +896,10 @@ func (c *client) processMsg(msg []byte) {
 			qsubs := r.qsubs[i]
 			index := c.cache.prand.Intn(len(qsubs))
 			sub := qsubs[index]
-			mh := c.msgHeader(msgh[:si], sub)
-			c.deliverMsg(sub, mh, msg)
+			if sub != nil {
```

Comment:
I guess unrelated to dynamic buffers, but is it not raising a question about wether this is a normal situation (`sub` being `nil`)?